model/project_ref.go

```diff
@@ -1226,13 +1231,39 @@ func (t *PatchTriggerDefinition) Validate(parentProject string) error {
 	if !utility.StringSliceContains([]string{"", AllStatuses, evergreen.PatchSucceeded, evergreen.PatchFailed}, t.Status) {
 		return errors.Errorf("invalid status: %s", t.Status)
 	}
-	_, regexErr := regexp.Compile(t.VariantRegex)
-	if regexErr != nil {
-		return errors.Wrap(regexErr, "invalid variant regex")
-	}
-	_, regexErr = regexp.Compile(t.TaskRegex)
-	if regexErr != nil {
-		return errors.Wrap(regexErr, "invalid task regex")
+
+	for _, specifier := range t.TaskSpecifiers {
+		if (specifier.VariantRegex != "" || specifier.TaskRegex != "") && specifier.PatchAlias != "" {
+			return errors.New("can't specify both a regex set and a patch alias")
+		}
+
+		if specifier.VariantRegex != "" {
+			_, regexErr := regexp.Compile(specifier.VariantRegex)
+			if regexErr != nil {
+				return errors.Wrapf(regexErr, "invalid variant regex '%s'", specifier.VariantRegex)
+			}
+		}
+
+		if specifier.TaskRegex != "" {
+			_, regexErr := regexp.Compile(specifier.TaskRegex)
+			if regexErr != nil {
+				return errors.Wrapf(regexErr, "invalid task regex '%s'", specifier.TaskRegex)
+			}
+		}
+
+		if specifier.PatchAlias != "" {
+			var aliases []ProjectAlias
+			aliases, err = FindAliasesForProject(t.ChildProject)
```

Comment:
You can use FindAliasInProject and pass in the specifier.PatchAlias

Reply:
done